### ISSUE TITLE
[Topic] WordPress Blocks (Can I use)

### ISSUE THREAD

**Issue by rianrietveld** [wpaccessibility/wp-a11y-docs#294]
- [x] I've confirmed that this document is not already in the [Docs Project](https://github.com/orgs/wpaccessibility/projects/3)

**Title**
WP Blocks
In the section WordPress and Accessibility

**Description**
Give info about which core blocks are accessible, where you can find the a11y issues with specific blocks and how to report issues.

**Reason for Adding**
We have a place for info about WP Themes and plugins, but not for blocks. How to find out which native blocks you can use safely?
This question was raised during the WPLDN meetup in Februari 2026.

## Proposed setup

We list the state of accessibility of core blocks. A "Can I use" list.
The data can be taken from Joe's [Block index](https://blocks.joedolson.com/) and other review data the A11y team has. 

We can create per block a separate page with:
- name block
- date last tested for accessibility
- status (Meets WCAG 2.2 AA / Needs work / Needstesting )
- link to the related a11y discussions or reviews on GitHub if there are any.
- link to audit?
- addition text, whatever is worth mentioning

## Discussion

As wpaccessibility.org is not a WordPress site, adding fully working examples are not possible.
So the examples on [blocks.joedolson.com/](https://blocks.joedolson.com/) are valuable. How do we keep both in sync? Or is there another easy way to point to examples of blocks.

Discussions about the accessibility of a block should live on the [Gutenberg repo, labelled focus accessibility](https://github.com/WordPress/gutenberg/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22%5BFocus%5D%20Accessibility%20(a11y)%22}) and not on this documentation site. 

The content of [Guidelines for Accessibility Testing of Block Editor Blocks & Patterns](https://blocks.joedolson.com/testing-guidance/) can be added as a separate topic to [Test for accessibility](https://wpaccessibility.org/docs/testing/).

About the [Block Review: WCUS Contributor Day 2022](https://docs.google.com/spreadsheets/d/1KjA57v8pl9sVw0ssuYCjX8z1rW7v-tG65ofHpjp8DAY/edit?gid=0#gid=0), we have a project for this in this GitHub organisation: [Gutenberg blocks testing](https://github.com/orgs/wpaccessibility/projects/2). 

If we set this up once for the current, link to the page Guidelines for Accessibility Testing of Block Editor Blocks & Patterns and add this to the info on
[Accessibility tickets, tasks, and reports](https://make.wordpress.org/accessibility/handbook/get-involved/tickets-tasks-reports/) and [Get Started at Contributor Day](https://make.wordpress.org/accessibility/handbook/get-involved/getting-started-at-a-contributor-day/). We have an easy answer on "How can I help".

**Comment by amberhinds** [comment]
Sorry for the slow response! Circling back, I'm not certain about the usefulness of this doc exactly as proposed. 

I *believe* most or all core blocks can be used and they should all be mostly accessible on the front end when released. Thus, I think the "can I use" for blocks would always be "yes, you can use it as long as you use it smartly." I.e., A single disclosure block is fine; a group of disclosure blocks isn't correct and should be an accordion block instead. 

Now, this information may all be totally different if we're talking about accessibility in the editor.

Another concern I have is the difficulty of displaying working examples that are all current. It is safe to assume that many blocks on that site from 2022 are no longer representative of what that same block would be like if inserted into the editor today. When WordPress updates, it won't fix or change any instances of blocks that are already inserted — only new instances of those blocks. So someone would have to manually update the block site regularly to keep it current.

The best I could think of to keep current examples of all blocks without a ton of ongoing manual labor is setting up a playground function to build the site with the current version of WordPress and current block examples. There are issues with Playground from both an accessibility perspective and an ability to use automated testing tools, because it is an iframe, but that is the only way to get recent versions of all available blocks, I believe.

To be honest, rather than having separate documentation, I think it would be much better to add accessibility information to the existing [Core Blocks Reference doc.](https://developer.wordpress.org/block-editor/reference-guides/core-blocks/). It would be better to have one source of truth on blocks than two.

**Comment by rianrietveld** [comment]
Thanks for your feedback @amberhinds , this is super helpful. 

I agree, adding the a11y into with the block docs is best place. I will think about how we best can communicate which blocks, pattens, themes and plugins etc to use or how to test for it, without having to do a ton of work to set up and maintain it ourselves.

**Comment by joedolson** [comment]
There's a proposal to automated publishing block documentation on .org, and that would basically take the blocks `readme.md` file and generate docs from it and other factors. This gives us a viable path to publish block documentation as part of the main block documentation, which I think would give us what we really need.

See https://github.com/WordPress/gutenberg/pull/77612 and https://make.wordpress.org/core/2026/05/05/proposal-auto-generate-block-editor-handbook-docs-from-block-json/
